hooks/pre-compact.sh

```
@@ -0,0 +1,155 @@
+#!/usr/bin/env bash
+# hooks/pre-compact.sh
+# PreCompact hook — reads session state + memory files and injects them
+# as additionalContext so critical info survives context compaction.
+#
+# Also called by SessionStart (after compaction) to re-inject memory.
+
+set -euo pipefail
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+PLUGIN_ROOT="${CLAUDE_PLUGIN_ROOT:-${extensionPath:-$(cd "$SCRIPT_DIR/.." && pwd)}}"
```

Shellcheck (0.11.0)

[warning] PLUGIN_ROOT appears unused. Verify use (or export if used externally).

(SC2034)